synapse/storage/room.py

```diff
@@ -71,6 +71,17 @@ def get_room(self, room_id):
             RoomsTable.decode_single_result, query, room_id,
         )
 
+    def get_all_rooms(self):
+        """Retrieve all the rooms.
+
+        Returns:
+            A list of namedtuples containing the room information.
+        """
+        query = RoomsTable.select_statement()
+        return self._execute(
+            RoomsTable.decode_results, query,
+        )
```

Comment:
At some point it would be nice to remove Tables (and potentially replace it with something else)